pkg/config/config_template.yaml

```diff
@@ -21,6 +21,19 @@ api_key:
 #
 # dd_url: https://app.datadoghq.com
 
+## @param cloud_provider - string - optional - default: <empty>
+## This option restrict which endpoint will be used by the agent to retrieve host metadata. By default the agent will try
+## all cloud provider.
+## Possible values are:
+## "AWS"     AWS EC2, ECS/Fargate
+## "GCP"     Google Cloud Provider
+## "Azure"   Azure
+## "Alibaba" Alibaba
+## "none"    does not query any cloud metadata endpoints (falling back on system metadata). Using this setting on a cloud provider host may result in duplicated hosts in your account.
+## ""        default: queries all cloud metadata endpoints
+#
+# cloud_provider: ""
```

Comment:
nit: let's move this config option down a bit in this config file, since most users shouldn't need to configure it. For example above the `collect_ec2_tags` option (that'll also "group" it with the other metadata-related options)

Reply:
also, `cloud_provider_metadata` might be a better name actually (describes more accurately what the option actually does)

Reply:
I've moved the option to be just above collect_ec2 tags and renamed to cloud_provider_metadata